pipe.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2017 HyperHQ Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package main
+
+import "io"
+
+type inPipe struct {
+	s    shimAgent
+	c, p string
```

Comment:
I think it would be clearer if all these members had longer more meaningful names.